demos/image-demo.py

```diff
@@ -32,10 +32,10 @@ def build_image_plan(dataset):
     start_time = time.time()
 
     parser = argparse.ArgumentParser(description="Run a simple demo")
-    parser.add_argument("--no-cache", action="store_true", help="Do not use cached results", default=True)
+    parser.add_argument("cache", action="store_false", help="Use cached results", default=False)
```

Comment:
TODO: fix this to have `--cache` and keep `store_true`

Reply:
For more detail, `action="store_true"` makes the argument parser store a default value of `False` for the variable if the argument is not provided by the user (the name is confusing). Thus, we are somewhat redundantly specifying `action="store_true"` AND `default=False`.

But I am happy to have this redundancy as it makes the code a bit more explicit and clear, so if you make the changes specified in the previous comment that should be good.

More info: https://stackoverflow.com/questions/8203622/argparse-store-false-if-unspecified#:~:text=The%20store_true%20option%20automatically%20creates,line%20argument%20is%20not%20present.